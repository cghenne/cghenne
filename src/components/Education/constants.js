export const education = [
  {
    graduation_year: 2008,
    school: 'Paul Lambin Institute',
    location: 'Brussels, Belgium',
    description: `BSc Computer Science (web services, infographic
      and artificial intelligence)`
  },
  {
    graduation_year: 2005,
    school: 'High School "Lyc√©e de Berlaymont"',
    location: 'Waterloo, Belgium',
    description: `Sciences and mathematics`
  }
];

export type SchoolT = {
  graduation_year: number,
  school: string,
  location: string,
  description: string
};
