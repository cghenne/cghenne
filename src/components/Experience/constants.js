export const jobs = [
  {
    start_date: 'January 2018',
    end_date: null,
    position: 'Senior Full-Stack Engineer',
    company: 'LifeWorks',
    location: 'Toronto, CA',
    description: `LifeWorks transfered me to their Toronto office to direct the
      development support in the Eastern Time Zone, continue the evolution and
      innovation of the product and create opportunities to build out a North
      America-based development team.`,
    tech_stack: null
  },
  {
    start_date: 'July 2016',
    end_date: 'December 2017',
    position: 'Lead Front-End Engineer',
    company: 'LifeWorks',
    location: 'London, UK',
    description: `In addition to front-end development,
      I performed hiring, training, and mentoring
      for the front-end team. I was also involved in project
      development and oversight and have helped to improve the
      operating platform.`,
    tech_stack: null
  },
  {
    start_date: 'July 2015',
    end_date: 'July 2016',
    position: 'Front-End Engineer',
    company: 'LifeWorks',
    location: 'London, UK',
    description: `LifeWorks: A Global Employee Engagement
      Platform serving 50,000 companies with over 15 million users.`,
    tech_stack: `JavaScript, React, AngularJS, HTML, CSS, Sass,
      Redux, Flow, Node.js, Webpack, Gulp, NPM, Clojure, ClojureScript,
      Jenkins, Vagrant`
  },
  {
    start_date: 'May 2012',
    end_date: 'July 2015',
    position: 'Senior Web Developer',
    company: 'Vanilla Active',
    location: 'London, UK',
    description: `Vanilla Active is a digital agency offering services to
      agencies and brands. We designed and implemented promotional web
      sites directly with brands and worked closely with advertising
      and marketing agencies.`,
    tech_stack: 'HTML, CSS, JavaScript, AngularJS, ColdFusion, PHP, Wordpress',
    print_margin: true
  },
  {
    start_date: 'November 2011',
    end_date: 'March 2012',
    position: 'Excel Expert',
    company: 'Wild Media Ltd',
    location: 'London, UK',
    description: `Wild Media Ltd is a startup in the new
      media area. As an Excel expert and
      database manager, I was in charge of optimising
      and automating the logistics department.`,
    tech_stack: null
  },
  {
    start_date: 'July 2011',
    end_date: 'November 2011',
    position: 'Consultant',
    company: 'De Valck Consultants by Altran',
    location: 'Brussels, Belgium',
    description: `De Valck Consultants is part of Altran group,
      the European leader in innovation consulting.
      It is the dedicated brand to Quality Assurance
      for the Life Science industry. I worked as an
      IS Validation Engineer at Baxter Healthcare.`,
    tech_stack: null
  },
  {
    start_date: 'December 2008',
    end_date: 'June 2011',
    position: 'Web Developer',
    company: 'Jules Bordet Institute',
    location: 'Brussels, Belgium',
    description: `The BrEAST data centre is a specialised clinical
      trials unit created in order to conduct large,
      international Phase III studies in breast cancer
      aiming to register new drugs. I worked as a web
      and application developer, in IT support and
      maintenance, and in validation of computer systems
      developed in-house (GCP, 21 CRF part 11).`,
    tech_stack: 'HTML, CSS, JavaScript, PHP, Java, Oracle SQL, MySQL'
  }
];

export type JobT = {
  start_date: string,
  end_date: ?string,
  position: string,
  company: string,
  location: string,
  description: ?string,
  tech_stack: ?string,
  print_margin?: boolean
};
