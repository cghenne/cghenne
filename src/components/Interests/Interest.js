// @flow
import React from 'react';
import './Interests.css';

type Props = {
  label: string,
  description: string
};

const Interest = ({ label, description }: Props) => {
  return (
    <li>
      <span className="bold">{label}</span>: {description}
    </li>
  );
};

Interest.displayName = 'Interest';

export default Interest;
