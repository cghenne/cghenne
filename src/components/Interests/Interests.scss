@import '../../styles/utils.scss';

.interests {
  grid-area: interests;

  .bold {
    font-weight: $font-weight-bold;
  }
}
