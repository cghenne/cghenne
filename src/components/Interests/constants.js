export const interests = [
  {
    type: 'sports',
    label: 'Sports',
    description: 'Outdoor sports, squash, badminton'
  },
  {
    type: 'Music',
    label: 'Music',
    description: 'Piano, Guitar'
  },
  {
    type: 'other',
    label: 'Other',
    description: 'Travelling'
  }
];

export type InterestT = {
  type: string,
  label: string,
  description: string
};
