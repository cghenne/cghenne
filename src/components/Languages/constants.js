export const languages = [
  {
    type: 'french',
    label: 'French',
    level: 'Native'
  },
  {
    type: 'english',
    label: 'English',
    level: 'Fluent'
  },
  {
    type: 'dutch',
    label: 'Dutch',
    level: 'Basic'
  }
];

export type LanguageT = {
  type: string,
  label: string,
  level: string
};
