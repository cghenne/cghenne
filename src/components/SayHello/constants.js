import github from './assets/github.svg';
import linkedin from './assets/linkedin.svg';
import email from './assets/email.svg';
import phone from './assets/phone.svg';

export const contacts = [
  {
    type: 'github',
    icon: github,
    link: '/cghenne',
    url: 'https://github.com/cghenne',
    isPrivate: false
  },
  {
    type: 'linkedin',
    icon: linkedin,
    link: '/cathleen-ghenne',
    url: 'https://www.linkedin.com/in/cathleen-ghenne/',
    isPrivate: false
  },
  {
    type: 'email',
    icon: email,
    link: 'cathleen.ghenne@gmail.com',
    url: 'mailto:cathleen.ghenne@gmail.com',
    isPrivate: true
  },
  {
    type: 'phone',
    icon: phone,
    link: '+1 437 928 2032',
    url: 'tel:+14379282032',
    isPrivate: true
  }
];

export type ContactT = {
  icon: string,
  link: string,
  url: string,
  show: boolean
};
