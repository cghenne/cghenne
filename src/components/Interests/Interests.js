// @flow
import React from 'react';
import { interests } from './constants';
import type { InterestT } from './constants';
import Interest from './Interest';
import './Interests.scss';

const Interests = () => {
  return (
    <div className="box interests">
      <h3>Personal Interests</h3>
      <ul>
        {interests.map((interest: InterestT) => (
          <Interest {...interest} key={interest.type} />
        ))}
      </ul>
    </div>
  );
};

Interests.displayName = 'Interests';

export default Interests;
