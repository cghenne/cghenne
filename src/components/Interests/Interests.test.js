import React from 'react';
import { shallow } from 'enzyme';
import Interests from './Interests';
import { interests } from './constants';

describe('Interests', () => {
  const wrapper = shallow(<Interests />);

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render title', () => {
    expect(wrapper.find('h3').text()).toEqual('Personal Interests');
  });

  it('should render the box with a class of box and interests', () => {
    expect(wrapper.find('div').prop('className')).toEqual('box interests');
  });

  it('should render a list', () => {
    expect(wrapper.find('ul').exists()).toEqual(true);
  });

  it('should render a Interest component for each interest in the list of interests', () => {
    expect(wrapper.find('Interest').length).toEqual(interests.length);
  });
});
