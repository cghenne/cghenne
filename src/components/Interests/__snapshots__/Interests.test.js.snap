// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interests should render without crashing 1`] = `
<div
  className="box interests"
>
  <h3>
    Personal Interests
  </h3>
  <ul>
    <Interest
      description="Outdoor sports, squash, badminton"
      key="sports"
      label="Sports"
      type="sports"
    />
    <Interest
      description="Piano, guitar"
      key="Music"
      label="Music"
      type="Music"
    />
    <Interest
      description="Travelling, painting"
      key="other"
      label="Other"
      type="other"
    />
  </ul>
</div>
`;
